soc/nxp/kinetis/ke1xz/soc.c

```diff
@@ -81,31 +87,90 @@ static const scg_firc_config_t scg_firc_config = {
+
+static const scg_lpfll_config_t scg_lpfll_config = {
+	.enableMode = kSCG_LpFllEnable,
+	.div2       = TO_ASYNC_CLK_DIV(SCG_CLOCK_DIV(flldiv2_clk)),
+#if MHZ(48) == DT_PROP(SCG_CLOCK_NODE(lpfll_clk), clock_frequency)
+	.range      = kSCG_LpFllRange48M,
+#elif MHZ(72) == DT_PROP(SCG_CLOCK_NODE(lpfll_clk), clock_frequency)
+	.range      = kSCG_LpFllRange72M,
+#elif MHZ(96) == DT_PROP(SCG_CLOCK_NODE(lpfll_clk), clock_frequency)
+	.range      = kSCG_LpFllRange96M,
+#else
+#error Invalid SCG FLL clock frequency
+#endif
+	.trimConfig = NULL,
+};
+
+
+static void CLOCK_CONFIG_FircSafeConfig(const scg_firc_config_t *fircConfig)
 {
-	const scg_sys_clk_config_t scg_sys_clk_config_safe = {
-		.divSlow = kSCG_SysClkDivBy4,
-		.divCore = kSCG_SysClkDivBy1,
-		.src     = kSCG_SysClkSrcSirc
+	scg_sys_clk_config_t curConfig;
+	const scg_sirc_config_t scgSircConfig       = {.enableMode = kSCG_SircEnable,
+						.div2       = kSCG_AsyncClkDivBy2,
+						.range      = kSCG_SircRangeHigh};
+	scg_sys_clk_config_t sysClkSafeConfigSource = {
+		.divSlow = kSCG_SysClkDivBy4, /* Slow clock divider */
+		.divCore = kSCG_SysClkDivBy1, /* Core clock divider */
+		.src     = kSCG_SysClkSrcSirc /* System clock source */
 	};
-	scg_sys_clk_config_t current;
+	/* Init Sirc. */
+	CLOCK_InitSirc(&scgSircConfig);
+	/* Change to use SIRC as system clock source to prepare to change FIRCCFG register. */
+	CLOCK_SetRunModeSysClkConfig(&sysClkSafeConfigSource);
+	/* Wait for clock source switch finished. */
+	do {
+		CLOCK_GetCurSysClkConfig(&curConfig);
+	} while (curConfig.src != sysClkSafeConfigSource.src);
+
+	/* Init Firc. */
+	CLOCK_InitFirc(fircConfig);
+	/* Change back to use FIRC as system clock source in order to configure SIRC if needed. */
+	sysClkSafeConfigSource.src = kSCG_SysClkSrcFirc;
+	CLOCK_SetRunModeSysClkConfig(&sysClkSafeConfigSource);
+	/* Wait for clock source switch finished. */
+	do {
+		CLOCK_GetCurSysClkConfig(&curConfig);
+	} while (curConfig.src != sysClkSafeConfigSource.src);
+}
 
-	/* Configure SIRC */
-	CLOCK_InitSirc(&scg_sirc_config);
 
-	/* Temporary switch to safe SIRC in order to configure FIRC */
-	CLOCK_SetRunModeSysClkConfig(&scg_sys_clk_config_safe);
-	do {
-		CLOCK_GetCurSysClkConfig(&current);
-	} while (current.src != scg_sys_clk_config_safe.src);
-	CLOCK_InitFirc(&scg_firc_config);
+__weak void clk_init(void)
+{
+	scg_sys_clk_config_t current;
 
-	/* Only RUN mode supported for now */
+#if DT_NODE_HAS_STATUS_OKAY(SCG_CLOCK_NODE(sosc_clk))
+	/* Init SOSC according to board configuration. */
+	CLOCK_InitSysOsc(&scg_sosc_config);
+	CLOCK_SetXtal0Freq(scg_sosc_config.freq);
+#endif
+	/* Init FIRC. */
+	CLOCK_CONFIG_FircSafeConfig(&scg_firc_config);
+	/* Init SIRC. */
+	CLOCK_InitSirc(&scg_sirc_config);
+	/* Init LPFLL. */
+	CLOCK_InitLpFll(&scg_lpfll_config);
+	/* Finally init the App desired clock */
 	CLOCK_SetRunModeSysClkConfig(&scg_sys_clk_config);
 	do {
-		CLOCK_GetCurSysClkConfig(&current);
+	     CLOCK_GetCurSysClkConfig(&current);
```

Comment:
The additional indent to CLOCK_GetCurSysClkConfig() needs to be another <tab> and not spaces.